PREPARE-TIMES-NZ/scripts/stage_1_prep_raw_data/extract_mbie_data.py

```
@@ -61,124 +52,149 @@ def get_mbie_electricity(SheetName, rowstart,rowend, category_name, unit, variab
-                          category_name = "Fuel",
-                          unit = "PJ",
-                          variable_name = "Annual net electricity generation")
-# save
-df.to_csv(f"{output_location}/mbie_ele_generation_pj.csv", index=False)
-
-#endregion
-
-#############################################################################
-#region ELE ONLY
-#############################################################################
-ele_only_gen = pd.read_excel(
-    f"{DATA_RAW}/external_data/mbie/electricity.xlsx",
-    sheet_name = "6 - Fuel type (GWh)",
-    usecols = "B:K", skiprows = 5, nrows = 51)
-
-
-# ensure year label correct 
-ele_only_gen = ele_only_gen.rename(columns = {"Unnamed: 1":  "Year"})
-# remove junk rows 
-ele_only_gen = ele_only_gen[~ele_only_gen["Year"].isna()]
-# remove provisional label
-ele_only_gen = ele_only_gen.drop(["Unnamed: 2"], axis = 1)
-
-# tidy column names 
-ele_only_gen = ele_only_gen.rename(
-    columns = {"Oil1": "Oil",
-               "Geo- thermal": "Geothermal"}
-    )
-# sensible shape 
-ele_only_gen = ele_only_gen.melt(
-    id_vars = "Year",
-    var_name = "Fuel",
-    value_name = "Value"
-    )
-# label 
-ele_only_gen["Unit"] = "GWh"
-ele_only_gen["Variable"] = "Electricity generation (no cogen)"
-ele_only_gen = ele_only_gen.rename(columns = {"Fuel":"FuelType"})
-# years as integers 
-ele_only_gen["Year"] = ele_only_gen["Year"].astype(int)
-
-
-# save
-ele_only_gen.to_csv(f"{output_location}/mbie_ele_only_generation.csv", index=False)
-#endregion
-
-#############################################################################
-#region ELE CAPACITY
-#############################################################################
-
 
-official_capacity = pd.read_excel(
+def get_mbie_gen_ele_only():
+    ele_only_gen = pd.read_excel(
+        f"{DATA_RAW}/external_data/mbie/electricity.xlsx",
+        sheet_name = "6 - Fuel type (GWh)",
+        usecols = "B:K", skiprows = 5, nrows = 51)
```

Could we be more defensive here - making fewer assumptions about structure of data (or at least validating that our assuptions are correct)? Would be good to fail early with helpful errors where appropriate (Are we expecting to re-use these functions for future MBIE data releases? Perhaps we need to timestamp the dataset?)

E.g. thoughts along the lines of putting in some more assertion statements
- REQUIRED_COLS present in the sheet
- expected number of rows in the filtered data
- centralizing 'magic ranges' in a lookup table at the start
- Possibly adding unit tests